datamodel/high/base/schema_proxy.go

```diff
@@ -14,6 +14,14 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// Package-level cache for high-level schemas
+var highSchemaCache = struct {
```

Comment:
we would want this cache scoped to the document otherwise in long running applications or between different documents we may be caching things incorrectly

But also not clearing the cache when we should